### ISSUE TITLE
evaluate_accuracy代码中的n是什么意思

### ISSUE THREAD

**Issue by uesrs** [PaddlePaddle/Paddle#54769]
### 请提出你的问题 Please ask your question

def evaluate_accuracy(data_iter,net):
    acc_sum ,n=0.0, 0
    with paddle.no_grad(): #在使用准确率评估模型效果期间，需要关闭模型的梯度追踪功能
        for batch_id, data in enumerate(data_iter()):
            X,y = data #读取一条数据
            X = paddle.to_tensor(X)
            y = paddle.to_tensor(y)
            net.eval() # 评估模式，这会关闭丢弃曾
            predicts, acc = net(X,y) #在通过向前传播过程得到预测值的同时计算准确率
            acc_sum += acc
            net.train() # 改回训练模式
            n = n+1
            # print(n)
    return acc_sum/n

**Comment by zhangbo9674** [comment]
您好，未在 develop 分支的 PaddlePaddle/Paddle repo 查找到 evaluate_accuracy，麻烦提供一下上述代码的具体位置

**Comment by uesrs** [comment]
> 

import pandas as pd
import numpy as np
import paddle
import paddle.fluid as fluid
import paddle.nn.functional as F
#%%
class NN_paddle(paddle.nn.Layer):
    def __init__(self, in_channels, out_channels):
        super(NN_paddle, self).__init__()

        self.fc1 = paddle.nn.Linear(in_channels, 32)
        self.do1 = paddle.nn.Dropout(0.3)
        self.fc2 = paddle.nn.Linear(32, 16)
        self.do2 = paddle.nn.Dropout(0.1)
        self.fc3 = paddle.nn.Linear(16, out_channels)

    def forward(self, X, label=None):

        outputs1 = F.relu(self.fc1(X))
        outputs1 = self.do1(outputs1)
        outputs2 = F.relu(self.fc2(outputs1))
        outputs2 = self.do2(outputs2)
        outputs3 = F.softmax(self.fc3(outputs2))

        if label is not None:
            acc = paddle.metric.accuracy(input=outputs3, label=label)
            return outputs3, acc
        else:
            return outputs3
#%%
data = pd.read_csv(r'clean_data_movie.csv').drop(['Unnamed: 0','Title'],axis=1)
#%%
def train_test_split(data_split, train_size_rate):
    assert train_size_rate<1 and train_size_rate>0, 'train_size_rate should be greater than 0 and smaller than 1'
    data_split = data_split.sample(frac=1).reset_index(drop=True)
    train_data = data_split.iloc[:int(len(data_split)*train_size_rate),:]
    test_data = data_split.iloc[int(len(data_split)*train_size_rate):,:]

    return train_data, test_data
#%%
def load_data(data_load, labels_name, batch_size):
    data_load = data_load.sample(frac=1).reset_index(drop=True)

    labels = data_load[labels_name]
    data_ready_to_load = data_load.drop(labels_name, axis=1)

    assert len(data_ready_to_load) == len(labels), \
        "length of train_imgs({}) should be the same as train_labels({})".format(len(data_load), len(labels))

    def data_generator():

        data_list = []
        labels_list = []

        for i in list(data_ready_to_load.index):
            data = np.array(data_ready_to_load.iloc[i,:]).astype('float32')
            label = np.array(labels[i]).astype('int64')
            label = np.reshape(labels[i], [1])

            data_list.append(data)
            labels_list.append(label)

            if len(data_list) == batch_size:
                yield np.array(data_list), np.array(labels_list)

                data_list = []
                labels_list = []

            if len(data_list) > 0:
                yield np.array(data_list), np.array(labels_list)

    return data_generator
#%%
#训练设置
def evaluate_accuracy(data_iter,net):
    acc_sum ,n=0.0, 0
    with paddle.no_grad(): #在使用准确率评估模型效果期间，需要关闭模型的梯度追踪功能
        for batch_id, data in enumerate(data_iter()):
            X,y = data #读取一条数据
            X = paddle.to_tensor(X)
            y = paddle.to_tensor(y)
            net.eval() # 评估模式，这会关闭丢弃曾
            predicts, acc = net(X,y) #在通过向前传播过程得到预测值的同时计算准确率
            acc_sum += acc
            net.train() # 改回训练模式
            n = n+1
            # print(n)
    return acc_sum/n
#%%
def train(model, train_data, test_data, labels_name, batch_size, lr, use_gpu):

    paddle.set_device('gpu:0') if use_gpu else paddle.set_device('cpu')

    model.train()

    train_loader = load_data(train_data, labels_name, batch_size)
    test_loader = load_data(test_data, labels_name, batch_size)

    opt = paddle.optimizer.Adam(learning_rate=lr, parameters=model.parameters(), weight_decay=0.001)

    epoch_num = 50
    for epoch_id in range(epoch_num):
        for batch_id, data in enumerate(train_loader()):
            X, y = data
            X = paddle.to_tensor(X)
            y = paddle.to_tensor(y)

            predicts, train_acc = model(X, y)

            loss = F.cross_entropy(predicts, y)
            avg_loss = paddle.mean(loss)

            test_acc = evaluate_accuracy(test_loader, model)

            if batch_id % 10 == 0:
                print('epoch:{}, batch:{}, loss is:{}, train_acc is:{}'.format(epoch_id, batch_id, avg_loss.numpy(), train_acc.numpy(), test_acc.numpy()))

                avg_loss.backward()
                opt.step()
                opt.clear_grad()

    paddle.save(model.state_dict(), './mnist.pdparams')
#%%
model = NN_paddle(36, 2)
train_data, test_data = train_test_split(data, 0.7)
train(model, train_data, test_data, 'Success', 100, 1e-3, False)

----这是所有的代码，谢谢大佬

**Comment by zhangbo9674** [comment]
你好，上述代码中，n 的含义可能只是记录一下 evaluate 的轮次，另外，这份代码是 paddle 官方提供的么，可否提供一下具体的位置？
